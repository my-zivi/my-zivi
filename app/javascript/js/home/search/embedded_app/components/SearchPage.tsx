import { Configure, InstantSearch, Stats } from 'react-instantsearch-dom';
import CustomAutocomplete from 'js/home/search/embedded_app/components/CustomAutocomplete';
import React, { JSX } from 'preact/compat';
import { SearchClient } from 'algoliasearch';
import qs from 'qs';
import MapSearchView from 'js/home/search/embedded_app/components/MapSearchView';
import TilesSearchView from 'js/home/search/embedded_app/components/TilesSearchView';

const HITS_PER_PAGE = 20;

type SearchState = Record<string, unknown>;
type Props = { searchClient: SearchClient };
type State = { searchState: SearchState };

const searchStateToUrl = (searchState: SearchState) => (searchState ? `?${qs.stringify(searchState)}` : '');
const urlToSearchState = (search: string) => qs.parse(search.slice(1));

class SearchPage extends React.Component<Props, State> {
  constructor() {
    super();

    this.state = {
      searchState: urlToSearchState(window.location.search),
    };
  }

  private onSearchStateChange: (searchState: SearchState) => void = (searchState) => {
    window.history.replaceState(null, null, searchStateToUrl(searchState));

    this.setState({ searchState });
  };

  render(): JSX.Element {
    const { searchClient } = this.props;
    const defaultRefinement = new URLSearchParams(window.location.search).get('q');

    return (
      <InstantSearch
        searchClient={searchClient}
        onSearchStateChange={this.onSearchStateChange}
        searchState={this.state.searchState}
        indexName="JobPosting"
      >
        <Configure hitsPerPage={HITS_PER_PAGE} clickAnalytics />
        <div className="search-main">
          <div className="hero">
            <div className="container">
              <div className="hero-content w-100">
                <div className="row">
                  <div className="col-xl-1" />
                  <div className="col"><CustomAutocomplete defaultRefinement={defaultRefinement} /></div>
                  <div className="col-xl-1" />
                </div>
              </div>
            </div>
          </div>

          <MapSearchView />
          {/*<TilesSearchView />*/}
        </div>
      </InstantSearch>
    );
  }
}

export default SearchPage;
