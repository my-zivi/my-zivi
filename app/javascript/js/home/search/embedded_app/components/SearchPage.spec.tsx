import { shallow } from 'enzyme';
import React from 'preact/compat';
import SearchPage from './SearchPage';
import { mockClient } from '../../../../tests/algolia/mocks';

describe('SearchPage', () => {
  it('renders correctly', () => {
    expect(
      shallow(<SearchPage searchClient={mockClient} />).find('.search-main').first(),
    ).toMatchSnapshot();
  });
});
