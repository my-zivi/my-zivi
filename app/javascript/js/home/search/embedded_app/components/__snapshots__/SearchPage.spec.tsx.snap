// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage renders correctly 1`] = `
<InstantSearch
  indexName="JobPosting"
  refresh={false}
  searchClient={
    Object {
      "addAlgoliaAgent": [Function],
      "addApiKey": [Function],
      "appId": "appid",
      "assignUserID": [Function],
      "assignUserIDs": [Function],
      "clearCache": [Function],
      "clearDictionaryEntries": [Function],
      "copyIndex": [Function],
      "copyRules": [Function],
      "copySettings": [Function],
      "copySynonyms": [Function],
      "deleteApiKey": [Function],
      "deleteDictionaryEntries": [Function],
      "destroy": [Function],
      "generateSecuredApiKey": [Function],
      "getApiKey": [Function],
      "getAppTask": [Function],
      "getDictionarySettings": [Function],
      "getLogs": [Function],
      "getSecuredApiKeyRemainingValidity": [Function],
      "getTopUserIDs": [Function],
      "getUserID": [Function],
      "hasPendingMappings": [Function],
      "initAnalytics": [Function],
      "initIndex": [Function],
      "initRecommendation": [Function],
      "listApiKeys": [Function],
      "listClusters": [Function],
      "listIndices": [Function],
      "listUserIDs": [Function],
      "moveIndex": [Function],
      "multipleBatch": [Function],
      "multipleGetObjects": [Function],
      "multipleQueries": [Function],
      "multipleSearchForFacetValues": [Function],
      "removeUserID": [Function],
      "replaceDictionaryEntries": [Function],
      "restoreApiKey": [Function],
      "saveDictionaryEntries": [Function],
      "search": [Function],
      "searchDictionaryEntries": [Function],
      "searchForFacetValues": [Function],
      "searchUserIDs": [Function],
      "setDictionarySettings": [Function],
      "transporter": Object {
        "headers": Object {
          "content-type": "application/x-www-form-urlencoded",
          "x-algolia-api-key": "apikey",
          "x-algolia-application-id": "appid",
        },
        "hosts": Array [
          Object {
            "accept": 3,
            "protocol": "http",
            "url": "example.com",
          },
        ],
        "hostsCache": Object {
          "clear": [Function],
          "delete": [Function],
          "get": [Function],
          "set": [Function],
        },
        "logger": Object {
          "debug": [Function],
          "error": [Function],
          "info": [Function],
        },
        "queryParameters": Object {},
        "read": [Function],
        "requester": Object {
          "send": [Function],
        },
        "requestsCache": Object {
          "clear": [Function],
          "delete": [Function],
          "get": [Function],
          "set": [Function],
        },
        "responsesCache": Object {
          "clear": [Function],
          "delete": [Function],
          "get": [Function],
          "set": [Function],
        },
        "timeouts": Object {
          "connect": 2,
          "read": 5,
          "write": 30,
        },
        "userAgent": Object {
          "add": [Function],
          "value": "Algolia for JavaScript (4.9.1); Node.js (12.22.1)",
        },
        "write": [Function],
      },
      "updateApiKey": [Function],
      "waitAppTask": [Function],
    }
  }
  stalledSearchDelay={200}
>
  <ConnectorWrapper
    hitsPerPage={20}
  />
  <div
    className="hero"
  >
    <div
      className="container"
    >
      <div
        className="hero-content w-100"
      >
        <ConnectorWrapper
          defaultRefinement={null}
        />
      </div>
    </div>
  </div>
  <div
    className="container mt-6"
  >
    <div
      className="d-flex justify-content-start justify-content-lg-end mb-2"
    >
      <div
        className="text-muted mr-1"
      >
        <ConnectorWrapper
          translations={
            Object {
              "stats": [Function],
            }
          }
        />
      </div>
      <ConnectorWrapper />
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-lg-9 order-1 order-lg-0"
      >
        <section
          className="jobs-hits pt-0"
        >
          <ConnectorWrapper />
        </section>
      </div>
      <div
        className="col-12 col-lg-3 order-0 order-lg-1"
      >
        <RefinementsPanel />
      </div>
    </div>
  </div>
</InstantSearch>
`;
